lib/response.js

```diff
@@ -417,71 +417,72 @@ function patch(Response) {
+            // res.send is called with an Object instead of calling res.json
+            if (!type && typeof body === 'object' && !Buffer.isBuffer(body)) {
+                type = 'application/json';
+            }
 
-        // Derive type if not provided by the user
-        type = type || self.req.accepts(self.acceptable);
-
-        // Check to see if we could find a content type to use for the response.
-        if (!type) {
-            return formatterError(
-                self,
-                new errors.NotAcceptableError({
-                    message:
-                        'could not find suitable content-type to use ' +
-                        'for the response'
-                })
-            );
-        }
+            // Derive type if not provided by the user
+            type = type || self.req.accepts(self.acceptable);
+
+            // Check to see if we could find a content type to use for the
+            // response.
+            if (!type) {
+                return formatterError(
+                    self,
+                    new errors.NotAcceptableError({
+                        message:
+                            'could not find suitable content-type to use ' +
+                            'for the response'
+                    })
+                );
+            }
 
-        type = type.split(';')[0];
+            type = type.split(';')[0];
 
-        if (!self.formatters[type] && type.indexOf('/') === -1) {
-            type = mime.lookup(type);
-        }
+            if (!self.formatters[type] && type.indexOf('/') === -1) {
+                type = mime.lookup(type);
+            }
 
-        formatter = self.formatters[type] || self.formatters['*/*'];
+            formatter = self.formatters[type] || self.formatters['*/*'];
 
-        if (self._charSet) {
-            type = type + '; charset=' + self._charSet;
-        }
+            if (self._charSet) {
+                type = type + '; charset=' + self._charSet;
+            }
 
-        // Update Content-Type header to the one originally set or to the type
-        // inferred from the most relevant formatter found.
-        self.setHeader('Content-Type', type);
+            // Update Content-Type header to the one originally set or to the
+            // type inferred from the most relevant formatter found.
+            self.setHeader('Content-Type', type);
 
-        if (!formatter) {
-            return flush(self, body);
+            if (formatter) {
+                // Finally, invoke the formatter and flush the request with it's
+                // results
+                return flush(self, formatter(self.req, self, body));
+            }
         }
 
-        // Finally, invoke the formatter and flush the request with it's results
-        return flush(self, formatter(self.req, self, body));
+        // if no formatting, assert that the value to be written is a string or
+        // a buffer, then send it.
+        assert.ok(
```

Comment:
I just realized that the formatters don't necessarily enforce a proper return value, so could we benefit from moving this assertion into `flush()`?

Reply:
👍